test/test_torch.py

```diff
@@ -16626,6 +16630,65 @@ def test_nextafter(self, device, dtype):
         expected = np.nextafter(a.cpu().numpy(), b.cpu().numpy())
         self.assertEqual(actual, expected, atol=0, rtol=0)
 
+    @dtypesIfCUDA(torch.float16, torch.float32, torch.float64)
+    @dtypesIfCPU(torch.bfloat16, torch.float32, torch.float64)
+    @dtypes(torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_i0_range1(self, device, dtype):
+        a = torch.rand(100, device=device).to(dtype) * 13.25
+        actual = torch.i0(a)
+        if dtype is torch.bfloat16:
+            a = a.to(torch.float32)
+        expected = scipy.special.i0(a.cpu().numpy())
+        if dtype is torch.bfloat16 or dtype is torch.float16:
+            expected = torch.from_numpy(expected).to(dtype)
+        self.assertEqual(actual, expected)
+
+        a = torch.rand(100, device=device).to(dtype) * -13.25
+        actual = torch.i0(a)
+        if dtype is torch.bfloat16:
+            a = a.to(torch.float32)
+        expected = scipy.special.i0(a.cpu().numpy())
+        if dtype is torch.bfloat16 or dtype is torch.float16:
+            expected = torch.from_numpy(expected).to(dtype)
+        self.assertEqual(actual, expected)
+
+    @dtypesIfCPU(torch.bfloat16, torch.float32, torch.float64)
+    @dtypes(torch.float32, torch.float64)
+    @precisionOverride({torch.bfloat16: 3e-2})
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_i0_range2(self, device, dtype):
+        a = torch.rand(1000, device=device).to(dtype) * 88.5
+        actual = torch.i0(a)
```

Comment:
Since the same block appears again here, you may want to refactor it into a helper function:

`def _i0_helper(t): ...`

Then you can just compute the tensors and call the helper function in these tests.